### PR TITLE
Migrates remaining Marionette components to React, removes Marionette dependency, removes Less (migrating to Tailwind)

### PR DESCRIPTION
**Pull request by andrewkfiedler**
Hard to note everything in this pull request, but the largest migrations were the upload area, notifications, the maps and their drawing / display of shapes.

Removes Marionette as a dependency 🥳 
Removes jquery-ui, a widget from jquery, and a date picker we no longer use as well.
Removes Less (step towards reducing our build complexity).  We're heading towards Tailwind for styling primarily.
Converts everything to typescript.
Moves us from cjs to es6.

Beyond that there is general deletion of code that was no longer used.

### CODE REVIEW

**Comment by rzwiefel** [comment]
🎉👏👏👏

**Review by mackncheesiest — APPROVED** [review]
image

**Review by vinamartin — APPROVED** [review]
Thank you 🙏

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

**Comment by andrewkfiedler** [comment]
build now

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/metacard-preview/metacard-preview.view.tsx [hunk 1]
```diff
@@ -13,81 +13,24 @@
...
-    />
+export const MetacardPreviewReact = ({
+  result,
+}: {
+  result: LazyQueryResult
+}) => {
+  return (
+    <Paper className="p-4 w-full h-full overflow-auto">
+      <span
+        className=" whitespace-pre-wrap"
```
Comment (glenhein) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/metacard-preview/metacard-preview.view.tsx:29]: Does it matter that there is extra whitespace in the string?

**Review by glenhein — APPROVED** [review]
Looked there was at least one new file that didn't have a copyright statement.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/metacard-preview/metacard-preview.view.tsx (hunk 1 above)
Comment (andrewkfiedler) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/metacard-preview/metacard-preview.view.tsx:29]: Nope

**Comment by andrewkfiedler** [comment]
> 

🤔 I would've thought spotless would catch that.

**Review by rececoffin — APPROVED** [review]

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/golden-layout/golden-layout.view.tsx [hunk 2]
```diff
@@ -263,262 +297,244 @@ export const getStringifiedDefaultLayout = () => {
...
+  lastConfig,
+}: {
+  goldenLayout: any
+  options: GoldenLayoutViewProps
+  lastConfig: React.MutableRefObject<any>
+}) {
+  // not sure we can do this if block with react, check if we need still
+  // if (this.isDestroyed) {
+  //   return
+  // }
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/golden-layout/golden-layout.view.tsx]: Can this be removed?

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/golden-layout/golden-layout.view.tsx [hunk 3]
```diff
@@ -263,262 +297,244 @@ export const getStringifiedDefaultLayout = () => {
...
+//   goldenLayoutInstance: any
+//   layout: any
+// }) {
+//   goldenLayoutInstance.root.contentItems.forEach((item: any) => {
+//     item.remove()
+//   })
+//   layout.content.forEach((item: any) => {
+//     goldenLayoutInstance.root.addChild(item)
+//   })
+// }
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/golden-layout/golden-layout.view.tsx]: Can this be removed?

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/golden-layout/golden-layout.view.tsx [hunk 4]
```diff
@@ -201,17 +241,16 @@ function registerComponent(
...
-
+// function isMaximised(contentItem: any) {
+//   if (contentItem.isMaximised) {
+//     return true
+//   } else if (contentItem.contentItems.length === 0) {
+//     return false
+//   } else {
+//     return _.some(contentItem.contentItems, isMaximised)
+//   }
+// }
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/golden-layout/golden-layout.view.tsx]: Can this be removed?

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/location-new/inputs/masked-text-field.tsx [hunk 5]
```diff
@@ -62,31 +63,40 @@ class MaskedTextField extends React.Component {
...
+            </div>
           ) : null}
           <MaskedInput
+            className="inline-block w-full whitespace-no-wrap flex-shrink overflow-hidden"
             value={value}
             keepCharPositions
-            onChange={(e) => {
-              this.props.onChange(e.target.value)
+            onChange={(e: any) => {
+              ;(this.props as any).onChange(e.target.value)
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/location-new/inputs/masked-text-field.tsx:89]: Was this semi colon an accident or do I just not know this syntax? I see it in a couple other places too 🤔 I assume it doesn't break anything, but just curious if it has a purpose.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/location-new/location.tsx [hunk 6]
```diff
@@ -165,7 +155,7 @@ const LocationInput = (props: LocationInputPropsType) => {
   }, [props])
 
   return (
-    <div>
+    <div className="">
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/location-new/location.tsx]: Doesn't really matter, but empty className

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/result-selector/query-feed.tsx [hunk 7]
```diff
@@ -172,7 +172,7 @@ const QueryStatus = ({
           Filter
         </HeaderCell>
       </tr>
-      <tbody className="is-list">
+      <tbody className="">
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/result-selector/query-feed.tsx]: Empty className

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/result-selector/query-feed.tsx [hunk 8]
```diff
@@ -272,7 +272,7 @@ const QueryFeed = ({ selectionInterface }: Props) => {
             <div className="relative">
               <Button
                 data-id="heartbeat-button"
-                className="details-view is-button"
+                className=""
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/result-selector/query-feed.tsx]: Empty className

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/cesium.view.tsx [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+// } catch (err) {
+//   console.error(err)
+//   this.$el.addClass('not-supported')
+//   setTimeout(() => {
+//     this.switchTo2DMap()
+//   }, 10000)
+//   this.endLoading()
+// }
+
+// featureDetection.addFailure('cesium')
```
Comment (sgalla) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/cesium.view.tsx]: Can this be removed?

**Review by sgalla — APPROVED** [review]

**Comment by pvargas** [comment]
Hero successful. However, given the number of changes, I'd recommend that someone else also hero it, so I won't not be tagging it as heroed. Up to you, though.

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/location-new/inputs/masked-text-field.tsx (hunk 5 above)
Comment (andrewkfiedler) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/location-new/inputs/masked-text-field.tsx:89]: 🤔 Prettier puts it automatically in a few places.  Looks like it's a consequence of our semicolon free style that we use.

Basically they are making sure the previous code doesn't get executed with (this.props) as an argument when this gets minified.  

https://stackoverflow.com/questions/7145514/whats-the-purpose-of-starting-semi-colon-at-beginning-of-javascript

**Comment by andrewkfiedler** [comment]
build now

**Pull request merged**